### ISSUE TITLE
Consider adopting Semantic Versioning

### ISSUE THREAD

**Issue by kossnocorp** [tdlib/td#367]
Hey, first of all, thank you for all your great work! I have never been working with an API that works well 24/7. Huge respect for that!

There are things that could be improved. In the latest release, you introduced a breaking change that might break existing code: https://github.com/tdlib/td/blame/master/CHANGELOG.md#L39

Please, consider adopting [Semantic Versioning](https://semver.org/). While I don't believe in bulletproof conventions and everyone should still read changelogs carefully, yet changing of the major version would make it easier.

**Comment by levlam** [comment]
I've just commented on Semantic Versioning here: https://github.com/tdlib/td/pull/362#issuecomment-422046221.

In details, TDLib JSON API, which is supposed to be used from the most programming languages, is loosely typed and conforms to Semantic Versioning. It has stable C ABI, which is changed only in backward compatible way, so MAJOR version is kept 1. There can be changes in underlying JSON objects, but these objects can be processed based on their content and the value of the "version" option. In fact, mentioned renaming of `Address` to `ShippingAddress` has no impact on JSON API, because that class has no subclasses and so its name is irrelevant. For example, it isn't used at all in the conversion of incoming requests from JSON objects to native requests.

From the other side, TDLib native C++/C#/Java interfaces are strongly typed class-based interfaces and are changed very-very often. Almost every such change is backward incompatible, so we would usually need to increase MAJOR version and MINOR/PATCH versions always will be near 0. It will eventually lead to a version hell. We could change MAJOR version, only for public releases, but in the meanwhile TDLib is used for Bot API, Android Telegram X updates and these versions have no chance to adhere to Semantic Versioning anyway. That's why we require [exact version match](https://github.com/tdlib/td/blob/101aa73f132d5124d8da45bae11a29c423420072/CMakeLists.txt#L802) for native C++ interface.

Semantic Versioning is a very good standard, but it is better suitable for long-living, general purpose, feature complete libraries then for a libraries which supposed to be often updated and for which the latest version supposed to be used. It is very unlikely that any application or its dependencies will depend on two different TDLib versions. It is also highly unrecommended to use old TDLib versions, because they lack support of important modern Telegram features, and server-side changes can make unstable some previously well-worked functionality.

To sum up, we can move in the future from `1.3.0` versioning scheme to `3.0` versioning scheme, but there is no chance that we will be able to use MINOR/PATCH version components in that scheme.

**Comment by levlam** [comment]
It is a good point. We definitely need to add to README that different TDLib versions should be treated as having incompatible API by default.
